source/io/module_eggShell.js

```diff
@@ -572,20 +571,30 @@
             return EggShell_GetArrayDimLength(Module.eggShell.v_userShell, vi, path, dim);
         };
 
-        Module.eggShell.resizeArray = publicAPI.eggShell.resizeArray = function (vi, path, newDimensionSizes) {
-            var int32Byte = 4;
-            var rank = newDimensionSizes.length;
-            var newLengths = Module._malloc(rank * int32Byte);
-
-            for (var i = 0; i < rank; i += 1) {
-                Module.setValue(newLengths + (i * int32Byte), newDimensionSizes[i], 'i32');
+        Module.eggShell.resizeArray = publicAPI.eggShell.resizeArray = function (valueRef, newDimensions) {
+            if (!Array.isArray(newDimensions)) {
+                throw new Error('Expected newDimensions to be an array of dimension lengths, instead got: ' + newDimensions);
             }
-
-            var success = EggShell_ResizeArray(Module.eggShell.v_userShell, vi, path, rank, newLengths);
-
-            Module._free(newLengths);
-
-            return success;
+            var stack = Module.stackSave();
+            var newDimensionsLength = newDimensions.length;
+            var dimensionsPointer = Module.stackAlloc(newDimensionsLength * LENGTH_SIZE);
+            var i, currentDimension;
+            for (i = 0; i < newDimensionsLength; i += 1) {
```

Comment:
Is a linting rule in `.eslintrc.js`. Why is preferred? Because `++` are subject to auto-semicolon insertion. Checkout eslint [docs](https://github.com/eslint/eslint/blob/master/docs/rules/no-plusplus.md). If we really don't like it, there's also the exception `"allowForLoopAfterthoughts": true`